### PR TITLE
test: add unit tests for services/session_parsers/claude_code.py

### LINKED ISSUE #1441: Add unit tests for services/session_parsers/claude_code.py
**Background**

services/session_parsers/claude_code.py parses Claude Code JSONL session transcripts into normalized frontend events, but currently has no dedicated parser test coverage.

**What needs to be done**

Create tests/test_claude_code_session_parser.py.

**Cover:**

User prompt parsing
Assistant response parsing
Thinking blocks
Tool use events
Tool result merging
System events
Attachment events
Invalid JSON handling
Meta event skipping

**Acceptance Criteria**

Major event types are covered by tests
Edge cases are covered
pytest tests/test_claude_code_session_parser.py -v passes
Existing test suite continues to pass

### PR DESCRIPTION
**Pull request by Madhumidha-S**
## Purpose / Description
Adds unit testing for the **Claude Code JSONL session parser** to ensure transcript lines are correctly parsed into normalised frontend events.

## Fixes
* Fixes #1441 

## Approach
Created a new test suite (`tests/test_claude_code_session_parser.py`) utilising `pytest`. Modelled against the existing session parser tests, the suite injects raw JSONL text samples through `parse_rows`.
- **Tool Result Merging:** Validated that tool result events correctly merge back into their parent tool use events, while orphan results safely skip without crashing.
- **Complex Turn Sequences:** Validated turns containing sequential combinations of thinking blocks, tool usages, and text responses.
- **Resilience:** Ensured malformed JSON, integer content types, and missing fields (like empty timestamps falling back to DB `ingested_at` sentinels) are handled gracefully.
- **Metrics & Metadata:** Verified accurate extraction of token usage metrics and system events, alongside the proper skipping of noisy events (e.g., `meta`, `agent-setting`).

## How Has This Been Tested?
- `uv run pytest tests/test_claude_code_session_parser.py -v` (all 23 tests passed).
- Verified linter rules via `uv run ruff check --fix`.
- Verified the global test suite continues to pass (`make test`).

## Checklist
- [x] You have a descriptive commit message with a short title (first line, max 50 chars).
- [x] You have commented your code, particularly in hard-to-understand areas
- [x] You have performed a self-review of your own code
- [ ] UI changes: include screenshots of all affected screens (in particular showing any new or changed strings)

## AI Assistance
_Was generative AI tooling used to co-author this PR?_
- [x] Yes(Please Specify the tool): Google Gemini
- [x] Was the generated code manually reviewed and tested? Yes

### CODE REVIEW

File: tests/test_claude_code_session_parser.py [hunk 1]
```diff
@@ -0,0 +1,218 @@
...
+import sys
+from pathlib import Path
+
+sys.path.insert(0, str(Path(__file__).parent.parent / "observal-server"))
+
+from services.session_parsers.claude_code import parse_rows
+
+# ── Sample Claude Code JSONL lines ────────────────────────────────────────────
+
+USER_SIMPLE = '{"type":"user","message":{"content":"Hello world"},"timestamp":"2026-06-01T10:00:00.000Z"}'
```
Comment (Haz3-jolt) [inline tests/test_claude_code_session_parser.py]: See these don't really follow the claude code jsonL pattern. If you do use. Claude code, copy over a small section containing tool calls and stuff. Copy that over and remove any PII. These don't follow that pattern. 

If you don't have access to claude code, please pick another issue.

**Review by Haz3-jolt — CHANGES_REQUESTED** [review]

File: tests/test_claude_code_session_parser.py (hunk 1 above)
Comment (Madhumidha-S) [reply tests/test_claude_code_session_parser.py]: Thanks for the review!! I'll update the fixtures to match actual Claude Code JSONL data.
